src/dropbox.js

```diff
@@ -1035,7 +950,6 @@ function hookSync(rs) {
   rs.sync.sync = function () {
     return this.dropbox.fetchDelta.apply(this.dropbox, arguments).
       then(rs._dropboxOrigSync, function (err) {
-        rs._emit('error', new Sync.SyncError(err));
```

Comment:
I forgot to mention in the commit message, but I removed the error event since the SyncError is not thrown when using WireClient either. The SyncError was causing the remoteStorage widget to display this error message, hide the sync button and basically becoming unusable unless the page was reloaded. However i had a second look now just to be sure and i realized the SyncError is not thrown when using WireClient because there is an actual bug in sync.js. In the following code, the resolve function is not bound to `this`, so a silent error that `this.remoteStorage` is undefined is thrown instead.

```
      return this.dealWithFailure(path, action, status).then(function () {
        this.remoteStorage._emit('error', error);
        throw error;
      });
```

So i guess sync.js needs the actual fixing, and errors in Dropbox's `fetchDelta ` should still be emitted, however i realized that instead of returning the rejection after emitting the error, the function should actually call original sync, otherwise there won't be any rescheduling done and periodic sync will stop.

Reply:
Good catch. This is actually fixed in my open refactoring PR, because it changes the function definition to a fat-arrow one, where `this` is bound to the enclosing context: https://github.com/remotestorage/remotestorage.js/pull/1147/files#diff-b5ce4044fcc6267228c6281fd7483b4cR871